src/tvheadend/HTSPVFS.cpp

```diff
@@ -93,30 +96,70 @@ void HTSPVFS::Close ( void )
   m_offset = 0;
   m_fileId = 0;
   m_path   = "";
+  m_eofOffsetSecs = -1;
+  m_pauseTime = 0;
+  m_paused = false;
+  m_isRealTimeStream = false;
 }
 
-ssize_t HTSPVFS::Read ( unsigned char *buf, unsigned int len )
+ssize_t HTSPVFS::Read ( unsigned char *buf, unsigned int len, bool inprogress )
 {
   /* Not opened */
   if (!m_fileId)
     return -1;
 
-  /* Read */
-  ssize_t read = SendFileRead(buf, len);
-
-  /* Update */
-  if (read > 0)
-    m_offset += read;
+  /* Tvheadend may briefly return 0 bytes when playing an in-progress recording at end-of-file
+     we'll retry 50 times with 10ms pauses (~500ms) before giving up */
+  int tries = inprogress ? 50 : 1;
+  ssize_t read = 0;
 
+  for (int i = 1; i <= tries; i++)
+  {
+    read = SendFileRead(buf, len);
+    if (read > 0)
+    {
+      m_offset += read;
+      return read;
+    }
+    else if (i < tries)
+      usleep(10000);
```

Comment:
Is usleep available on Windows? If not, we need a portable solution before we can merge this.

Reply:
We could use https://en.cppreference.com/w/cpp/thread/sleep_for

Reply:
I was just looking at the same page.

This seems to compile
```cpp
#include <chrono>
#include <thread>
...
else if (i < tries)
  std::this_thread::sleep_for(std::chrono::milliseconds(10));
```

Reply:
yaeh, I guess this will do the trick.

Reply:
Change pushed....still need to run through my tests...

Reply:
sure, ping me if you're done